src/backend/columnar/columnar_metadata.c

```diff
@@ -1206,11 +1247,20 @@ ReadDataFileStripeList(uint64 storageId, Snapshot snapshot)
 
 
 /*
- * BuildStripeMetadata builds a StripeMetadata object from given datumArray.
+ * BuildStripeMetadata builds a StripeMetadata object from given heap tuple.
  */
 static StripeMetadata *
-BuildStripeMetadata(Datum *datumArray)
+BuildStripeMetadata(HeapTuple heapTuple)
 {
+	Relation columnarStripes = RelationIdGetRelation(ColumnarStripeRelationId());
```

Comment:
won't we have a `Relation` at hand already on every call site for this function? Why not pass it along?

alternatively, I believe the `columnar.stripe` table only has fixed-length-fields. Could we make a `FormData`-esque struct to extract the fields? We can look at uses of `FormData_columnar_options` as how that is done from a `heapTuple`.